ignite/services/network/profile.go

```diff
@@ -0,0 +1,188 @@
+	)
+	if cosmoserror.Unwrap(err) == cosmoserror.ErrNotFound {
+		return sdk.Coins{}, ErrObjectNotFound
+	} else if err != nil {
+		return sdk.Coins{}, err
+	}
+	return res.Balances, nil
+}
+
+// MainnetAccountBalance returns the spn account or vesting account balance by address from SPN
+func (n Network) MainnetAccountBalance(
+	ctx context.Context,
+	launchID uint64,
+	address string,
+) (balance sdk.Coins, vesting sdk.Coins, err error) {
+	acc, err := n.GenesisAccount(ctx, launchID, address)
+	switch {
+	case err == ErrObjectNotFound:
+		accVest, err := n.VestingAccount(ctx, launchID, address)
+		if err != nil && err != ErrObjectNotFound {
+			return nil, nil, err
+		}
+		balance = accVest.TotalBalance
+		vesting = accVest.Vesting
+	case err != nil:
+		return nil, nil, err
+	default:
+		balance = acc.Coins
+	}
+	return
+}
+
+// CampaignAccountBalance returns the spn mainnet account or mainnet
+// vesting account balance by address from SPN
+func (n Network) CampaignAccountBalance(
+	ctx context.Context,
+	campaignID uint64,
+	address string,
+) (shares campaigntypes.Shares, vestingShares campaigntypes.Shares, err error) {
+	acc, err := n.MainnetAccount(ctx, campaignID, address)
+	switch {
+	case err == ErrObjectNotFound:
+		accVest, err := n.MainnetVestingAccount(ctx, campaignID, address)
+		if err != nil && err != ErrObjectNotFound {
+			return nil, nil, err
+		}
+		shares = accVest.TotalShares
+		vestingShares = accVest.Vesting
+	case err != nil:
+		return nil, nil, err
+	default:
+		shares = acc.Shares
+	}
+	return
+}
+
+// Profile returns the address profile info
+func (n Network) Profile(ctx context.Context, campaign bool, campaignID uint64) (networktypes.Profile, error) {
+	address := n.account.Address(networktypes.SPN)
+	vouchers, err := n.Balances(ctx, address)
+	if err != nil {
+		return networktypes.Profile{}, err
+	}
+
+	var (
+		shares             campaigntypes.Shares
+		vestingShares      campaigntypes.Shares
+		chainShares        []networktypes.ChainShare
+		chainVestingShares []networktypes.ChainShare
+	)
+	if campaign {
+		campaignChains, err := n.CampaignChains(ctx, campaignID)
+		if err == ErrObjectNotFound {
+			return networktypes.Profile{}, fmt.Errorf("invalid campaign id %d", campaignID)
+		} else if err != nil {
+			return networktypes.Profile{}, err
+		}
+		shares, vestingShares, err = n.CampaignAccountBalance(ctx, campaignID, address)
+		if err != nil && err != ErrObjectNotFound {
+			return networktypes.Profile{}, err
+		}
+		for _, chain := range campaignChains.Chains {
+			balance, vesting, err := n.MainnetAccountBalance(ctx, chain, address)
+			if err != nil && err != ErrObjectNotFound {
+				return networktypes.Profile{}, err
+			}
+			chainShares = append(chainShares, networktypes.ChainShare{
+				LaunchID: chain,
+				Shares:   balance,
+			})
+			chainVestingShares = append(chainVestingShares, networktypes.ChainShare{
+				LaunchID: chain,
+				Shares:   vesting,
+			})
+		}
+	}
+
+	var p networktypes.IProfile
```

Comment:
Why do we need an `IProfile` interface? Can we just use `Profile` here? Also the naming for interface is a bit off, not Go like.